Branch-SDK/Branch-SDK/BNCStrongMatchHelper.m

```diff
@@ -95,69 +155,121 @@ + (NSURL *)getUrlForCookieBasedMatchingWithBranchKey:(NSString *)branchKey redir
+
+    IMP implementation = class_getMethodImplementation(templateClass, selector);
+    if (!implementation) return NO;
+
+    class_addMethod(subclass, selector, implementation, typeEncoding);
+    return YES;
+}
+
+- (BOOL) willLoadViewControllerWithURL:(NSURL*)matchURL {
+    if (self.primaryWindow) return NO;
+
+    //  Dynamically subclass the SFSafariViewController if available.
+    //  This allows us to compile and link to an app that doesn't
+    //  include SafariServices, but is also able to compile and link
+    //  when it is.
+
+    Class SFSafariViewControllerClass = NSClassFromString(@"SFSafariViewController");
+    if (!SFSafariViewControllerClass) return NO;
+
+    Class BNCMatchViewControllerSubclass = NSClassFromString(@"BNCMatchViewController_Safari");
+    if (!BNCMatchViewControllerSubclass) {
+        //  The class isn't registered.  Create it:
+
+        BNCMatchViewControllerSubclass =
+            objc_allocateClassPair(SFSafariViewControllerClass, "BNCMatchViewController_Safari", 0);
+        if (!BNCMatchViewControllerSubclass) return NO;
+
+        BOOL fail = NO;
+        fail |= ![self subclass:BNCMatchViewControllerSubclass selector:@selector(becomeFirstResponder)];
+        fail |= ![self subclass:BNCMatchViewControllerSubclass selector:@selector(canBecomeFirstResponder)];
+        fail |= ![self subclass:BNCMatchViewControllerSubclass selector:@selector(nextResponder)];
+        if (fail) return NO;
+
+        objc_registerClassPair(BNCMatchViewControllerSubclass);
     }
+
+    self.matchViewController = [[BNCMatchViewControllerSubclass alloc] initWithURL:matchURL];
+    if (!self.matchViewController) return NO;
+    
+    self.primaryWindow = [[UIApplication sharedApplication] keyWindow];
+    self.matchViewController.delegate = self;
+    self.matchViewController.view.frame = self.primaryWindow.bounds;
+
+    [self.primaryWindow.rootViewController addChildViewController:self.matchViewController];
+    [self.primaryWindow insertSubview:self.matchViewController.view atIndex:0];
+    [self.matchViewController didMoveToParentViewController:self.primaryWindow.rootViewController];
+
+    return YES;
 }
 
-- (BOOL)shouldDelayInstallRequest {
-    return _shouldDelayInstallRequest;
+- (void) unloadViewController {
+    //NSLog(@"unloadViewController");
+    [self.matchViewController willMoveToParentViewController:nil];
+    [self.matchViewController.view removeFromSuperview];
+    [self.matchViewController removeFromParentViewController];
+     self.matchViewController.delegate = nil;
+     self.matchViewController = nil;
+     self.primaryWindow = nil;
+
+    [BNCPreferenceHelper preferenceHelper].lastStrongMatchDate = [NSDate date];
+    self.shouldDelayInstallRequest = NO;
+    self.requestInProgress = NO;
 }
 
+- (void)safariViewController:(SFSafariViewController *)controller
+      didCompleteInitialLoad:(BOOL)didLoadSuccessfully {
+    NSLog(@"Safari Did load. Success: %d.", didLoadSuccessfully);
```

Comment:
The problem is that this is only place you're referencing a class that can't exist when SafariServices.framework is not present. I saw your note that you tested without SafariServices.framework, but I just don't understand how this could technically build without the class present?